### PR TITLE
feat: whitespace in interpolation

### PR DESCRIPTION
**Pull request by mgechev**
First part of https://github.com/mgechev/codelyzer/issues/314

### CODE REVIEW

File: src/angularWhitespaceRule.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            expr.trim().length,
+            error, getReplacements(text, absolutePosition)));
+      }
+    }
+  }
+}
+
+export class Rule extends Lint.Rules.AbstractRule {
+  public static metadata: Lint.IRuleMetadata = {
+    ruleName: 'templates-use-public-rule',
```
Comment (wKoza) [inline src/angularWhitespaceRule.ts]: Wrong copy / paste ?

Comment (mgechev) [inline src/angularWhitespaceRule.ts]: Thanks for the catch :)

File: src/angularWhitespaceRule.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+const InterpolationOpen = Config.interpolation[0];
+const InterpolationClose = Config.interpolation[1];
+const InterpolationNoWhitespaceRe = new RegExp(`${InterpolationOpen}\\S(.*?)\\S${InterpolationClose}`);
+const InterpolationExtraWhitespaceRe =
+  new RegExp(`${InterpolationOpen}\\s\\s(.*?)\\s${InterpolationClose}|${InterpolationOpen}\\s(.*?)\\s\\s${InterpolationClose}`);
+
+const getReplacements = (text: ast.BoundTextAst, absolutePosition: number) => {
+  const expr: string = (text.value as any).source;
+  const internalStart = expr.indexOf(InterpolationOpen);
+  const internalEnd = expr.lastIndexOf(InterpolationClose);
```
Comment (wKoza) [inline src/angularWhitespaceRule.ts]: useless

File: test/angularWhitespaceRule.spec.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        \`
+      })
+      class Bar {}
+      `;
+      assertSuccess('angular-whitespace', source);
+    });
+  });
+
+  describe('failure', () => {
+    it('should not fail when no decorator is set', () => {
```
Comment (wKoza) [inline test/angularWhitespaceRule.spec.ts]: Wrong test description

File: test/angularWhitespaceRule.spec.ts [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+      assertAnnotated({
+        ruleName: 'angular-whitespace',
+        message: 'Missing whitespace in interpolation; expecting {{ expr }}',
+        source
+      });
+    });
+  });
+
+  describe('replacements', () => {
+    it('fixes negated pipes', () => {
```
Comment (wKoza) [inline test/angularWhitespaceRule.spec.ts]: Wrong test description

**Comment by mgechev** [comment]
@wKoza implemented the second rule for whitespace. Would you review?

**Comment by mgechev** [comment]
@wKoza fixed. Will merge today.

**Pull request merged**